sim/midas/src/main/scala/midas/passes/fame/EmitAndWrapRAMModels.scala

```diff
@@ -89,12 +89,15 @@ class ReadPort(val anno: ModelReadPort, val ports: Seq[Port]) extends IsMemoryPo
   def attachChannels(readCmd: Decoupled, readData: Decoupled): Seq[Statement] = {
     val iSignals = Seq(anno.addr, anno.en)
     val iValids  = iSignals.map(valid)
+    def iReady(i: ReferenceTarget) = {
+      And.reduce(readCmd.ready +: iSignals.filterNot(_ == i).map(valid))
```

Comment:
Wait, there's a reduce method on `And`. Tight.

Reply:
You were already using it, lol